tests/beta/tapes/test_cv_param_shift.py

```diff
@@ -0,0 +1,829 @@
+            qml.Rotation(0.543, wires=[0])
+            qml.Squeezing(0.543, 0, wires=[0])
+            expval(qml.P(0))
+
+        assert tape._grad_method(0) == "F"
+        assert tape._grad_method(1) == "A"
+        assert tape._grad_method(2) == "A"
+
+    def test_non_gaussian_operation(self):
+        """Test that a non-Gaussian operation succeeding
+        a differentiable Gaussian operation results in
+        numeric differentiation."""
+
+        with CVParamShiftTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.Rotation(1.0, wires=[1])
+            # Non-Gaussian
+            qml.Kerr(1.0, wires=[1])
+            expval(qml.P(0))
+            expval(qml.X(1))
+
+        # First rotation gate has no succeeding non-Gaussian operation
+        assert tape._grad_method(0) == "A"
+        # Second rotation gate does no succeeding non-Gaussian operation
+        assert tape._grad_method(1) == "F"
+        # Kerr gate does not support the parameter-shift rule
+        assert tape._grad_method(2) == "F"
+
+        with CVParamShiftTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.Rotation(1.0, wires=[1])
+            # entangle the modes
+            qml.Beamsplitter(1.0, 0.0, wires=[0, 1])
+            # Non-Gaussian
+            qml.Kerr(1.0, wires=[1])
+            expval(qml.P(0))
+            expval(qml.X(1))
+
+        # After entangling the modes, the Kerr gate now succeeds
+        # both initial rotations
+        assert tape._grad_method(0) == "F"
+        assert tape._grad_method(1) == "F"
+        assert tape._grad_method(2) == "F"
+
+    def test_probability(self):
+        """Probability is the expectation value of a
+        higher order observable, and thus only supports numerical
+        differentiation"""
+        with CVParamShiftTape() as tape:
+            qml.Rotation(0.543, wires=[0])
+            qml.Squeezing(0.543, 0, wires=[0])
+            probs(wires=0)
+
+        assert tape._grad_method(0) == "F"
+        assert tape._grad_method(1) == "F"
+        assert tape._grad_method(2) == "F"
+
+    def test_variance(self):
+        """If the variance of the observable is first order, then
+        parameter-shift is supported. If the observable is second order,
+        however, only finite-differences is supported."""
+
+        with CVParamShiftTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            var(qml.P(0))  # first order
+
+        assert tape._grad_method(0) == "A"
+
+        with CVParamShiftTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            var(qml.NumberOperator(0))  # second order
+
+        assert tape._grad_method(0) == "F"
+
+        with CVParamShiftTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.Rotation(1.0, wires=[1])
+            qml.Beamsplitter(0.5, 0.0, wires=[0, 1])
+            var(qml.NumberOperator(0))  # second order
+            expval(qml.NumberOperator(0))

```

Comment:
Hmm, would we perhaps like `var` and `expval` in such cases to be independent? :thinking:

Reply:
How do you mean @antalszava?

Reply:
It seems to me that we have finite differences here just because the first measurement is a variance. The second measurement returning the expectation value could be done using the parameter shift rule. Could we treat the two cases independently? (As they affect the same wire they would actually stand for two separate circuit evaluations)

Reply:
Oh, this is a typo! it should be 
```suggestion
            var(qml.NumberOperator(0))  # second order
            expval(qml.NumberOperator(1))
```
😆 

Essentially, this is just testing that the existance of the `var` forces all parameters in the circuit that directly contribute to the variance to swap to finite-differences. Even if they occur on wire 1.